restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -1144,15 +1166,25 @@ private boolean pathsMatch(String path1, String path2) {
                             return pathsMatch(rowPath + r.getString("leaf"), filterFieldPath);
                           })
                       .findFirst();
-              return fieldRow.isPresent() && allFiltersMatch(fieldRow.get(), inMemoryFilters);
+              return fieldRow.isPresent()
+                  && allFiltersMatch(fieldRow.get(), inMemoryFilters, numericBooleans);
             })
         .flatMap(x -> x.stream())
         .collect(Collectors.toList());
   }
 
-  private boolean allFiltersMatch(Row row, List<FilterCondition> filters) {
+  private Boolean getBooleanFromRow(Row row, String colName, boolean numericBooleans) {
+    if (row.isNull("bool_value")) return null;
+    if (numericBooleans) {
```

Comment:
Looks like this flag is use in many places... Would it make sense to convert it to an field on the `DocumentService` object?

Reply:
Yea I could definitely do that - before I had `!db.supportsSecondaryIndex()` inlined everywhere and this was a step cleaner than that; what you suggested is even better xD

Reply:
There is one issue with this - technically a DataStore is only in scope after an auth token is processed during a request, but the DocumentService is a singleton that is created on service startup. So in practice, there's actually a new DataStore object on every single request, and that's why I didn't make it a field on the service.

There's definitely a way to not repeat myself so much here...